app/models/stat.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+class Stat
+  def initialize(name, show_in_ui: false, expose_via_api: false, &block)
+    @name = name
+    @show_in_ui = show_in_ui
+    @expose_via_api = expose_via_api
+    @block = block
+  end
+
+  attr_reader :name, :expose_via_api, :show_in_ui
+
+  def calculate
+    begin
+      return @block.call.transform_keys { |key| build_key(key) }
+    rescue StandardError => err
+      Discourse.warn_exception(
+        err,
+        message: "Unexpected error when collecting #{@name} About stats.",
+      )
+    end
+    {}
+  end
+
+  def self.all_stats
+    calculate(_all_stats)
+  end
+
+  def self.exposable_stats
+    calculate(_exposable_stats)
+  end
+
+  private
+
+  def build_key(key)
+    "#{@name}_#{key}".to_sym
+  end
+
+  def self._all_stats
+    core_stats.concat(plugin_stats)
+  end
+
+  def self.calculate(stats)
+    stats.map { |stat| stat.calculate }.reduce(Hash.new, :merge)
+  end
+
+  def self.core_stats
+    [
+      Stat.new("topics", show_in_ui: true, expose_via_api: true) { Statistics.topics },
+      Stat.new("posts", show_in_ui: true, expose_via_api: true) { Statistics.posts },
+      Stat.new("users", show_in_ui: true, expose_via_api: true) { Statistics.users },
+      Stat.new("active_users", show_in_ui: true, expose_via_api: true) { Statistics.active_users },
+      Stat.new("likes", show_in_ui: true, expose_via_api: true) { Statistics.likes },
+    ]
+  end
+
+  def self._exposable_stats
+    _all_stats.select { |stat| stat.expose_via_api }
+  end
```

Comment:
It's not clear from the name "exposable_stats" that's this is due to the API

Reply:
This makes sense, I used the name `api_stats` in early versions, but then decided to use `exposable`. Let's use api_stats 👍.